Framework/Parallel/inc/MantidParallel/Collectives.h

```diff
@@ -68,6 +69,19 @@ void gather(const Communicator &comm, const T &in_value, int root) {
         "Parallel::gather on root rank without output argument.");
   }
 }
+
+template <typename T>
+void all_to_all(const Communicator &comm, const std::vector<T> &in_values,
+                std::vector<T> &out_values) {
+  int tag{0};
```

Comment:
Is there any significance of this constant? I think boost mpi has an `any_tag`

Reply:
It does (http://www.boost.org/doc/libs/1_54_0/doc/html/boost/mpi/any_tag.html), but "This constant may be used for the tag parameter of receive operations to indicate that a send with any tag will be matched by the receive." is not what we want, as far as I understand.

Reply:
Also this is code using the threading backend, not `boost::mpi`.